File: Projeto 4a/atividade3/InsertionSort.c
// Discente Maria Eduarda Guedes Alves

#include <stdio.h>
#include <stdlib.h>
#include <time.h>

#define N 100

void vetRound(int vet[], int n);
void printVet(int vet[], int n);
void InsertionSort(int vet[], int n);

int main() {
    double tGasto=0;
    clock_t t=0;
    int n=1;

    for (; tGasto<60000 ;) {
        n*=10; 
        int vet[n];
        vetRound(vet, n);

        t = clock();
        InsertionSort(vet, n);
        t = clock()-t;

        tGasto += ((double)t) / CLOCKS_PER_SEC * 1000;
        printVet(vet, n);
        printf("\nTempo gasto nesta chamada: %.6lf ms", tGasto);
        system("pause");
    }

    return 0;
}

void vetRound(int vet[], int n) {
    srand(time(NULL));

    for (int i=0; i<n; i++) {
        vet[i] = 1 + rand() % N;
    }
}

void printVet(int vet[], int n) {
    printf("[ ");
    for (int i = 0; i<n; i++)
        printf("%d ", vet[i]);
    printf("]\n");
}

void InsertionSort(int vet[], int n) {
    int i, j, pivo;

    for (i=1; i<n; i++) {
        pivo = vet[i];

        for (j = i-1; j>=0 && vet[j] > pivo; j--) {
            vet[j+1] = vet[j];
        }
        vet[j+1] = pivo;
    }
}

File: Projeto 4a/atividade3/InsertionSort.java
// Discente Maria Eduarda Guedes Alves

package atividade3;

import java.util.Random;
import java.util.Scanner;

public class InsertionSort {
    private static final int N = 100;

    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        double tGasto=0;
        long t;
        int n=1;

        for (; tGasto<60000; ) {
            n *= 10;
            int[] vet = new int[n];
            vetRound(vet, n);

            t = System.currentTimeMillis();
            InsertionSort(vet, n);
            t = System.currentTimeMillis() - t;

            tGasto += t;

            System.out.println();
            printVet(vet, n);
            System.out.printf("Tempo gasto nesta chamada: %.6f ms\n", tGasto);

            System.out.println("Pressione Enter para continuar...");
            scan.nextLine();
        }
        scan.close();
    }

    private static void vetRound(int vet[], int n) {
        Random rand = new Random();

        for (int i = 0; i < n; i++) {
            vet[i] = 1 + rand.nextInt(N);
        }
    }

    private static void printVet(int vet[], int n) {
        System.out.print("[ ");
        for (int i = 0; i < n; i++)
            System.out.print(vet[i] + " ");
        System.out.println("]");
    }

    private static void InsertionSort(int vet[], int n) {
        int i, j, pivo;

        for (i = 1; i < n; i++) {
            pivo = vet[i];

            for (j = i - 1; j >= 0 && vet[j] > pivo; j--) {
                vet[j + 1] = vet[j];
            }
            vet[j + 1] = pivo;
        }
    }
}
